x-pack/plugins/cases/public/components/filter_popover/index.tsx

```diff
@@ -88,16 +95,30 @@ export const FilterPopoverComponent = ({
       repositionOnScroll
     >
       <ScrollableDiv>
-        {options.map((option, index) => (
-          <EuiFilterSelectItem
-            checked={selectedOptions.includes(option) ? 'on' : undefined}
-            data-test-subj={`options-filter-popover-item-${option}`}
-            key={`${index}-${option}`}
-            onClick={toggleSelectedGroupCb.bind(null, option)}
-          >
-            {option}
-          </EuiFilterSelectItem>
-        ))}
+        {options.map((option, index) =>
+          typeof option === 'string' ? (
```

Comment:
How about we extract this functionality into a different component that specifically for filtering solutions?